src/loaders/compute_tools/tool_common.py

```diff
@@ -298,27 +299,20 @@ def parallel_single_execution(self, tool_callable: Callable[[str, str, Path, Pat
                  meta.get(loader_common_names.META_UNCOMPRESSED_FILE,
                           meta[loader_common_names.META_SOURCE_FILE]),
                  output_dir,
-                 self._program_threads,
+                 self._threads_per_tool_run,
                  self._debug))
 
         try:
-            self._execute(self._threads, tool_callable, args_list, start, False)
+            self._execute(tool_callable, args_list, start, False)
         finally:
-            if unzipped_files_to_delete:
-                print(f"Deleting {len(unzipped_files_to_delete)} unzipped files: {unzipped_files_to_delete[:5]}...")
-                for file in unzipped_files_to_delete:
-                    os.remove(file)
-
-        create_metadata_file(genomes_meta, batch_dir)
+            self._finalize_execution(unzipped_files_to_delete, genomes_meta, batch_dir)
```

Comment:
This seems like a logic error. With the previous set up, if _execute throws an error the unzipped files are cleaned up and the metadata file isn't created (as you would expect if there were errors). Now if an exception is thrown the metadata file will be created

Reply:
👍

Reply:
:+1: